config/confignet/confignet.go

```diff
@@ -36,20 +79,20 @@ type NetAddr struct {
 // Dial equivalent with net.Dialer's DialContext for this address.
 func (na *NetAddr) Dial(ctx context.Context) (net.Conn, error) {
 	d := net.Dialer{Timeout: na.DialerConfig.Timeout}
-	return d.DialContext(ctx, na.Transport, na.Endpoint)
+	return d.DialContext(ctx, string(na.Transport), na.Endpoint)
```

Comment:
nit: I think having an explicit `String` method would be better

Reply:
@mx-psi `String` won't play well with the consts unless we make the function a value receiver right? But then we'd be mixing value and reference receivers.

Reply:
Yup, makes sense, we can resolve this